packages/shared/lib/core/layer-2/utils/getSmartContractHexName.ts

```diff
@@ -0,0 +1,19 @@
+import { Blake2b } from '@iota/crypto.js'
+import { Converter } from '@iota/util.js'
+import { Buffer } from 'buffer'
+import { SimpleBufferCursor } from '../classes'
+
+export function getSmartContractHexName(name: string): number {
+    const SMART_CONTRACT_HEX_NAME_LENGTH = 4
+    const stringBytes = Converter.utf8ToBytes(name)
+    const hash = Blake2b.sum256(stringBytes)
+
+    for (let i = 0; i < hash.length; i += SMART_CONTRACT_HEX_NAME_LENGTH) {
+        const slice = hash.slice(i, i + SMART_CONTRACT_HEX_NAME_LENGTH)
+        const cursor = new SimpleBufferCursor(Buffer.from(slice))
+
+        return cursor.readUInt32LE()
+    }
```

Comment:
we return from this loop everytime at the first iteration, is this correct?

Reply:
yes you are right, I updated the code 🙏🏼